src/zftools/metagraph.pyx

```diff
@@ -31,14 +31,22 @@ from cpython.mem cimport (
 from zftools.fastqueue cimport FastQueueForBFS
 
 
+def verify_graph_is_compatible(sage_graph):
+    if sage_graph.is_directed():
+        raise ValueError("This is a directed graph.  Currently, directed graph zero forcing is not implemented.")
+    elif sage_graph.has_loops():
+        raise ValueError("This graph has loops.  Currently, looped zero forcing is not implemented.")
+
+
 def zero_forcing_set(sage_graph):
     cdef:
         ZFSearchMetagraph metagraph = ZFSearchMetagraph(sage_graph)
         frozenset start = frozenset()
         frozenset end = frozenset(
                 metagraph.to_relabeled_metavertex(sage_graph.vertices(sort=False))
                 )
-
+    
+    verify_graph_is_compatible(sage_graph)
```

Comment:
It seems like we have 2 choices regarding where to put this:
1. Leave it in `zero_forcing_set()`, in which case I'd say let's move this check to the very top of the function
2. Put it at the beginning of `ZFSearchMetagraph.__init__()`

I say perhaps we can go with 1. since it's what the user calls. Thoughts?

Reply:
Hmm!!  Wow, you're right.  I think for sure 2 is the right way to go.  I wish I had thought of that.  It is true that there are different requirements for the primal graph as a function of what variant of zero forcing is being computed, but each instance of a metagraph (which, again, will be an abstract class at some point) computes a specific variant and therefore needs a specific type of graph.  So it really needs to be switched.  I'll get to work on that.

Reply:
Now I am going the other way on this.  Or maybe going both ways at once?

I realized that eventually the helper functions down the road are going to have to be able to instantiate different subclasses of the metagraph class according to what form of forcing is called for.  And decisions about whether the graph is directed or not (for example) need to be made in order to figure that out.  So this logic has to go in the helper functions.  But it also makes sense for each subclass of metagraph to be checking whether it is appropriate for the primal graph on which it is instantiated.  So... maybe doing both (1) and (2) makes sense?

Running into this issue is the only reason I have not yet pushed any new commit to address this.

For now, maybe the best solution is to implement (1) first, and later on maybe also require subclasses to provide (2) as a check as well.

Reply:
Hm I'm not too much of a fan of duplicating the requirements checking if we can avoid it. What do you think about doing (2), then in the helper functions, we can do something like:
```python
try:
    mg = ZFSearchMetagraph(some_graph)
except SomeException as e:
   # handle accordingly
```

Reply:
That's an interesting idea!  I had to think on this a bit, since it amounts to saying that if a certain subclass can be instantiated, then this implies that that subclass is in fact the appropriate one to use.  At first this felt like an assumption with some danger to it, but the more that I think about it, then more I think this might be right.  In short, absent any revelations I haven't had in the past 10 minutes, this does seem like the way to go.